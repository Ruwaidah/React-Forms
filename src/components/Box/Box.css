.Box {
    margin: 4px;
}

.Box button {
    font-weight: bold;
    display: flex;
    justify-content: center;
    align-items: center;
    margin-top: 2px;
}