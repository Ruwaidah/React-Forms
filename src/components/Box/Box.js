import "./Box.css";
const Box = (props) => {
  return (
    <div className="Box">
      <div
        style={{
          background: props.box.color,
          width: props.box.width + "px",
          height: props.box.height + "px",
        }}
      ></div>
      <button
        style={{
          width: props.box.width + "px",
        }}
        onClick={() => props.deleteBox(props.idx)}
      >
        Delete
      </button>
    </div>
  );
};

export default Box;
