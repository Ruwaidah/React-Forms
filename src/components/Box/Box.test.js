import { render } from "@testing-library/react";
import Box from "./Box";

const box = {
  width: "200",
  height: "200",
  color: "#7d3f3f",
};
it("renders without crash", () => {
  render(<Box box={box} />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<Box box={box} />);
  expect(asFragment()).toMatchSnapshot();
});
