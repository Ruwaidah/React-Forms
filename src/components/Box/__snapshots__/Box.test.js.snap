// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
  >
    <div
      style="background: rgb(125, 63, 63); width: 200px; height: 200px;"
    />
    <button
      style="width: 200px;"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
